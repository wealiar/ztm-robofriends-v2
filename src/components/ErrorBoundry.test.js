import { shallow } from 'enzyme';
import React from 'react';
import ErrorBoundry from './ErrorBoundry';

let wrapper;
const testDiv = <div>{'love testing'}</div>;
const errorMessage = <h1>Ops. That is not good</h1>;

beforeEach(() => {
    wrapper = shallow(<ErrorBoundry>{testDiv}</ErrorBoundry>);
});

function FuncToSimulateError() {
    return null;
}

describe('ErrorBoundry smart component', () => {
    it('expect to render ErrorBoundry component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('when no error, expect to get my test div', () => {
        expect(wrapper.instance().render()).toEqual(testDiv);
    });

    it('when there is an error, expect to get error text', () => {
        wrapper.instance().setState({ hasError: true });
        expect(wrapper.instance().render()).toEqual(errorMessage);
    });

    it('catch an error and expect to get error text', () => {
        const wrapper2 = shallow(<ErrorBoundry><FuncToSimulateError /></ErrorBoundry>);
        const error = new Error('I\'m an error');
        wrapper2.find(FuncToSimulateError).simulateError(error);

        expect(wrapper2.instance().render()).toEqual(errorMessage);
    });

});
