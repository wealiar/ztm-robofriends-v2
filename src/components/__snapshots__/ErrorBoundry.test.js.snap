// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundry smart component expect to render ErrorBoundry component 1`] = `ShallowWrapper {}`;
